tests/securedrop/install.pm

```diff
@@ -0,0 +1,101 @@
+    # Borrowed from https://github.com/QubesOS/openqa-tests-qubesos/blob/7242736/tests/update2.pm#L121-L138
+
+    my ($self) = @_;
+    my $fname = $self->save_and_upload_log('rpm -qa', 'dom0-packages.txt');
+    my $packages = path('ulogs', $fname)->slurp;
+    $packages = join("\n", sort split(/\n/, $packages));
+    my $all_packages = "Dom0:\n" . $packages;
+    my $templates = script_output('qvm-ls --raw-data --fields name,klass');
+    foreach (sort split /\n/, $templates) {
+        next unless /Template/;
+        s/\|.*//;
+        $fname = $self->save_and_upload_log("qvm-run --no-gui -ap $_ 'rpm -qa; dpkg -l; pacman -Q; true'",
+                "template-$_-packages.txt", {timeout =>90});
+        $packages = path('ulogs', $fname)->slurp;
+        $packages = join("\n", sort split(/\n/, $packages));
+        $all_packages .= "\n" . $_ . ":\n" . $packages;
+        #assert_script_run("qvm-run --service -p $_ qubes.PostInstall", timeout => 90);
+        script_output("qvm-features $_", timeout => 90);
+        assert_script_run("qvm-shutdown --wait $_", timeout => 90);
+    }
+    path("sut_packages.txt")->spew($all_packages);
+}
+
+sub run {
+    my ($self) = @_;
+
+    $self->select_gui_console;
+    assert_screen "desktop";
+
+    x11_start_program('xterm');
+    send_key('alt-f10');  # maximize xterm to ease troubleshooting
+
+    curl_via_netvm;
+
+    # NOTE: These are done via qvm-run instead of gnome-terminal so that we
+    # can know in case they failed.
+    assert_script_run('qvm-run -p work -- gpg --keyserver hkps://keys.openpgp.org --recv-key "2359 E653 8C06 13E6 5295 5E6C 188E DD3B 7B22 E6A3"');
+    assert_script_run('qvm-run -p work -- "gpg --armor --export 2359E6538C0613E652955E6C188EDD3B7B22E6A3 > securedrop-release-key.pub"');
+    assert_script_run('qvm-run -p work -- sudo rpmkeys --import securedrop-release-key.pub');
+    assert_script_run('qvm-run -p work -- "echo -e \"[sd]\nenabled=1\nbaseurl=https://yum.securedrop.org/workstation/dom0/f37\nname=boostrap\"  | sudo tee /etc/yum.repos.d/securedrop-temp.repo"');
+    assert_script_run('qvm-run -p work -- dnf download -y securedrop-workstation-dom0-config');
+    assert_script_run('qvm-run -p work -- "rpm -Kv securedrop-workstation-dom0-config-*.rpm"');  # TODO confirm output is correct
+    assert_script_run('qvm-run -p work -- "cat /home/user/securedrop-workstation-dom0-config-*.rpm" > securedrop-workstation.rpm');
+    assert_script_run('sudo dnf -y install securedrop-workstation.rpm');
+    assert_script_run('echo {\"submission_key_fpr\": \"65A1B5FF195B56353CC63DFFCC40EF1228271441\", \"hidserv\": {\"hostname\": \"bnbo6ryxq24fz27chs5fidscyqhw2hlyweelg4nmvq76tpxvofpyn4qd.onion\", \"key\": \"FDF476DUDSB5M27BIGEVIFCFGHQJ46XS3STAP7VG6Z2OWXLHWZPA\"}, \"environment\": \"prod\", \"vmsizes\": {\"sd_app\": 10, \"sd_log\": 5}} | sudo tee /usr/share/securedrop-workstation-dom0-config/config.json');
```

Comment:
If we git cloned the sd-workstation repo (which we'll need to do to run the tests), we'd be able to copy this and the test key.

Reply:
At some point, you probably will want to control what gets installed via some openqa variable. See tests/update2.pm (among other things, it supports adding an extra repo with stuff to test - this is how we run tests on PR content before merging it). But I'd suggest to start with a single simple workflow and add extra flexibility later.